tools/base/fpgaconf/bitstream-tools.h

```diff
@@ -27,26 +27,77 @@
 // TODO: Remove metadata parsing code duplication by using
 // metadata parsing code in FPGA API
 
+#ifndef __BITSTREAM_TOOLS_H__
+#define __BITSTREAM_TOOLS_H__
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <uuid/uuid.h>
 #include <json-c/json.h>
 
-#ifndef __BITSTREAM_H__
-#define __BITSTREAM_H__
+#define METADATA_GUID                      "58656F6E-4650-4741-B747-425376303031"
+#define METADATA_GUID_LEN                   16
+#define GBS_AFU_IMAGE                       "afu-image"
+#define BBS_INTERFACE_ID                    "interface-uuid"
+#define PR_INTERFACE_ID                     "pr/interface_id"
+#define INTFC_ID_LOW_LEN                    16
+#define INTFC_ID_HIGH_LEN                   16
+#define GUID_LEN                            36
+#define AFU_NAME_LEN                        512
+
+// GBS json metadata
+// GBS version
+#define GBS_VERSION                                 "version"
+
+// AFU image
+#define GBS_AFU_IMAGE                               "afu-image"
+#define GBS_MAGIC_NUM                               "magic-no"
+#define BBS_INTERFACE_ID                            "interface-uuid"
+#define GBS_CLOCK_FREQUENCY_HIGH                    "clock-frequency-high"
+#define GBS_CLOCK_FREQUENCY_LOW                     "clock-frequency-low"
+#define GBS_AFU_POWER                               "power"
 
-#define METADATA_GUID "58656F6E-4650-4741-B747-425376303031"
-#define METADATA_GUID_LEN 16
-#define GBS_AFU_IMAGE "afu-image"
-#define BBS_INTERFACE_ID "interface-uuid"
+// AFU Clusters
+#define GBS_ACCELERATOR_CLUSTERS                    "accelerator-clusters"
+#define GBS_AFU_NAME                                "name"
+#define GBS_ACCELERATOR_TYPE_UUID                   "accelerator-type-uuid"
+#define GBS_ACCELERATOR_TOTAL_CONTEXTS              "total-contexts"
 
-#define PR_INTERFACE_ID "pr/interface_id"
-#define INTFC_ID_LOW_LEN 16
-#define INTFC_ID_HIGH_LEN 16
 
 #define PRINT_MSG printf
```

Comment:
Looks good!  I think you can remove these.

Reply:
get_fpga_interface_id() needs PR_INTERFACE_ID to access interface_id

Reply:
Was actually referring to PRINT_MSG and PRINT_ERR.  Not a biggee.  Thanks for all your work here!